src/test/java/htsjdk/samtools/cram/compression/rans/RansTest.java

```diff
@@ -49,107 +57,145 @@ public String toString() {
+        // TODO: More formatFlags values i.e, combinations of bit flags will be added later
+        return new Object[][]{
+                {ransNx16Encode, ransNx16Decode, new RANSNx16Params(0)} //RANSNx16 formatFlags(first byte) 0
+        };
+    }
+
+    public Object[][] getAllRansCodecs() {
+        // concatenate RANS4x8 and RANSNx16 codecs
+        return Stream.concat(Arrays.stream(getRans4x8Codecs()), Arrays.stream(getRansNx16Codecs()))
+                .toArray(Object[][]::new);
     }
 
-    @Test
-    public void testSizeRangeSmall() {
-        final byte[] data = randomBytesFromGeometricDistribution(1000, 0.01);
-        final ByteBuffer in = ByteBuffer.wrap(data);
-        for (int size = 4; size < data.length; size++) {
+    @DataProvider(name="allRansAndData")
+    public Object[][] getAllRansAndData() {
+        // this data provider provides all the testdata for all of RANS codecs
+        return TestNGUtils.cartesianProduct(getRansTestData(), getAllRansCodecs());
+    }
+
+    @DataProvider(name="allRansAndDataForTinySmallLarge")
+    public Object[][] getAllRansAndDataForTinySmallLarge() {
+        // this data provider provides Tiny, Small and Large testdata for all of RANS codecs
+        return TestNGUtils.cartesianProduct(getRansTestDataTinySmallLarge(), getAllRansCodecs());
+    }
+
+    @Test(dataProvider = "allRansAndDataForTinySmallLarge")
+    public void testSizeRangeTinySmallLarge(
+            final TestCaseWrapper tc,
+            final Integer lowerLimit,
+            final Integer upperLimit,
+            final RANSEncode ransEncode,
+            final RANSDecode ransDecode,
+            final RANSParams params){
+        final ByteBuffer in = ByteBuffer.wrap(tc.testArray);
+        for (int size = lowerLimit; size < upperLimit; size++) {
             in.position(0);
             in.limit(size);
-            roundTripForEachOrder(in);
+            ransRoundTrip(in, ransEncode, ransDecode, params);
         }
     }
 
-    @Test
-    public void testLargeSize() {
-        final int size = 100 * 1000 + 3;
-        final byte[] data = randomBytesFromGeometricDistribution(size, 0.01);
-        final ByteBuffer in = ByteBuffer.wrap(data);
-        for (int limit = size - 4; limit < size; limit++) {
-            in.position(0);
-            in.limit(limit);
-            roundTripForEachOrder(in);
-        }
+    @Test(dataProvider = "rans4x8")
+    public void testRans4x8BuffersMeetBoundaryExpectations(
+            final RANS4x8Encode ransEncode,
+            final RANS4x8Decode ransDecode,
+            final RANS4x8Params params) {
+        final int size = 1001;
+        final ByteBuffer compressed = ransBufferMeetBoundaryExpectations(size,ransEncode, ransDecode,params);
+        Assert.assertTrue(compressed.limit() > 10);
+        Assert.assertEquals(compressed.get(), (byte) params.getOrder().ordinal());
+        Assert.assertEquals(compressed.getInt(), compressed.limit() - 1 - 4 - 4);
```

Comment:
Same here - can we make constants with symbolic names for whatever the 1 and the 4 represent ?